cmake/Modules/GetVersionFromGit.cmake

```diff
@@ -0,0 +1,109 @@
+# GetVersionFromGit.cmake
+# Standalone script to retrieve versioning information from Git or .git_archival.txt.
+# Customizable for any project by setting variables before including this file.
+
+# Configurable variables:
+#   - VERSION_PREFIX: Prefix for version tags (default: "v").
+#   - VERSION_SUFFIX: Suffix for version tags (default: "[~+-]([a-zA-Z0-9]+)").
+#   - VERSION_REGEX: Regex to extract version (default: "(?[0-9]+\\.[0-9]+\\.[0-9]+)").
+#   - ARCHIVAL_FILE: Path to .git_archival.txt (default: "${CMAKE_SOURCE_DIR}/.git_archival.txt").
+#   - DESCRIBE_NAME_KEY: Key for describe name in .git_archival.txt (default: "describe-name: ").
+#   - COMMIT_HASH_KEY: Key for commit hash in .git_archival.txt (default: "commit: ").
+
+# Default Format Example:
+#   1.2.3 v1.2.3 v1.2.3-rc1
+
+set(VERSION_PREFIX "v" CACHE STRING "Prefix used in version tags")
+set(VERSION_SUFFIX "[~+-]([a-zA-Z0-9]+)" CACHE STRING "Suffix used in version tags")
+set(VERSION_REGEX "?([0-9]+\\.[0-9]+\\.[0-9]+)" CACHE STRING "Regex for extracting version")
+set(ARCHIVAL_FILE "${CMAKE_SOURCE_DIR}/.git_archival.txt" CACHE STRING "Path to .git_archival.txt")
+set(DESCRIBE_NAME_KEY "describe-name: " CACHE STRING "Key for describe name in .git_archival.txt")
+set(COMMIT_HASH_KEY "commit: " CACHE STRING "Key for commit hash in .git_archival.txt")
+
+
+# Combine prefix and regex
+set(VERSION_REGEX_WITH_PREFIX "^${VERSION_PREFIX}${VERSION_REGEX}")
+
+# Ensure Git is available
+find_package(Git REQUIRED)
```

Comment:
Is this actually required?  Have we tested on an image without git? It seems like there is a path to build with `GIT_FOUND` being false

Reply:
This has been fixed now. I just downloaded a .zip of the branch and the versioning logic appears to work correctly